src/Core/Grid/Query/CategoryQueryBuilder.php

```diff
@@ -195,6 +195,10 @@ private function getQueryBuilder(array $filters)
             }
 
             if ('id_category_parent' === $filterName) {
+                if ($this->isSearchRequestOnHomeCategory($filters)) {
```

Comment:
Thanks for the efforts in this complex matter. 
This `if` check in line 198 is what creates the problem raised here https://github.com/PrestaShop/PrestaShop/issues/20067
The system assumes that, if parent ID is sent in query string  (i.e key `id_category_parent` exists in `$filters` array), then automatically the merchant is not in home category.  So checking again if he is in Home just to skip the "WHERE" clause in query builder will put inconsistency in database results.
The "WHERE id_parent= X" is necessary whenever the `$filters['id_category_parent']` is set.
I am really interested in starting a PR about this complex matter, but I wanted to discuss with you about this choice before making first commit. Is there any specific reason for creating method `isSearchRequestOnHomeCategory`?